### ISSUE TITLE
Visualization

### ISSUE THREAD

**Issue by SSonnyboy** [jaewonalive/PeerAiD#2]
Could the author please provide the script for these visual diagrams used in the paper? For example, loss of landscape, feature representation, and semantic gradient

**Comment by jaewonalive** [comment]
Sorry for the late reply.

**Loss landscape.** For the loss landscape figure, I used the scripts from this repository: https://github.com/tomgoldstein/loss-landscape (it supports both 2D and 3D loss landscape plots).

**t-SNE.** For the t-SNE plot, I used `sklearn.manifold.TSNE`. We extracted a vector representation from the model, got flattened features and then ran t-SNE on the features:

```python
from sklearn.manifold import TSNE

tsne = TSNE(n_components=2, init='pca', random_state=0)
output_array_transformed = tsne.fit_transform(output_array_total)
```
**Semantic gradients**. For the semantic gradient–style visualization, I referred to the MadryLab robust-features codebase : https://github.com/MadryLab/robust-features-code/tree/master

In my implementation, I clip the values to within ±3 standard deviations, take the absolute value, sum across channels, and then apply min–max normalization to scale the resulting map to [0,1].
